### PR TITLE
feat(loader): update loader to adhere to the latest spec

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #4470 #8735 

## Summary

Updates the loader design to adhere to the design spec.

### Noteworthy changes

* simplified classes
* simplifies loader structure and animation
* remove obsolete `no-padding` attribute style
* drops `will-change` workaround based on perf testing
Screenshot 2024-11-22 at 3 44 11 PM
Screenshot 2024-11-22 at 3 44 17 PM

### CODE REVIEW

**Comment by jcfranco** [comment]
@SkyeSeitz @ashetland Could you please review the [screenshot diffs](https://www.chromatic.com/build?appId=6266d45509d7eb004aa254fb&number=6490)? 🙇

File: packages/calcite-components/src/components/loader/loader.scss [hunk 1]
```diff
@@ -11,140 +11,113 @@
...
-$loader-circumference: ($loader-scale - (2 * $stroke-width)) * 3.14159;
-
 @media (prefers-reduced-motion: reduce) {
   :root {
     --calcite-internal-duration-factor: 0;
   }
 }
 
 :host {
-  @apply relative mx-auto hidden items-center justify-center opacity-100;
```
Comment (driskull) [inline packages/calcite-components/src/components/loader/loader.scss:25]: Was loader hidden by default previously? Does the displays e2e test need to be updated?

File: packages/calcite-components/src/components/loader/loader.scss [hunk 2]
```diff
@@ -11,140 +11,113 @@
...
   --calcite-loader-size-inline: theme("spacing.6");
   --calcite-internal-loader-value-line-height: 1.71875rem; // 27.5px
 }
 
-:host([no-padding]) {
-  @apply py-0;
-}
-
-:host {
-  @apply flex;
```
Comment (driskull) [inline packages/calcite-components/src/components/loader/loader.scss:64]: oh ok it was hidden but then displayed flex here LOL

**Review by driskull — APPROVED** [review]
LGTM! 🎉

**Review by ashetland — APPROVED** [review]
🚀✨🎉

**Pull request merged**